src/Core/Generator/Sampler/CosmosExecutor.cs

```diff
@@ -53,7 +53,15 @@ public async Task<List<T>> ExecuteQueryAsync<T>(string query, Action<T?>? callba
                     string content = await sr.ReadToEndAsync();
 
                     using JsonDocument jsonDocument = JsonDocument.Parse(content);
-                    JsonElement root = jsonDocument.RootElement.GetProperty("Documents");
+
+                    JsonElement root = jsonDocument.RootElement;
+                    if (jsonDocument.RootElement.ValueKind != JsonValueKind.Array)
+                    {
+                        if (!jsonDocument.RootElement.TryGetProperty("Documents", out root))
+                        {
+                            root = jsonDocument.RootElement;
+                        }
+                    }
 
                     if (root.ValueKind == JsonValueKind.Array)
```

Comment:
Please consider renaming the root variable. 

At Line 66 it has a confusing double usage of the variable `root` which could either mean 

A. the jsonDocument's root element in 2 cases: 
1. when valuekind is array or
2. when we fail to get the Documents property if the valueKind is not array.

OR the `root` could also mean 
value of the Documents property when the jsonDocument's value kind is NOT Array.

Reply:
It is there to find out the root, in case of simple JSON, root would be actual root but if root has _Documents_ then, documents tag would become the root. It is more of a safety check. In cosmos db, we get documents under "Documents" tag.
Once root is finalized then, it checks if whatever is under that, is an array or not, then process the data accordingly.